app/Http/Controllers/Api/UserController.php

```diff
@@ -29,25 +44,36 @@ public function create()
     /**
      * Store a newly created resource in storage.
      *
-     * @param \Illuminate\Http\Request $request request create
+     * @param UserRegisterRequest $request request store data user
      *
      * @return \Illuminate\Http\Response
      */
-    public function store(Request $request)
+    public function store(UserRegisterRequest $request)
     {
-        $request = $request;
+        $user = $this->user->create($request->all());
+        if (!$user) {
+            return response()->json(['success' => false], Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+
+        return response()->json(['data' => $user, 'success' => true], Response::HTTP_OK);
     }
 
     /**
      * Display the specified resource.
      *
-     * @param int $id id user
+     * @param Request $request request get user information
      *
-     * @return \Illuminate\Http\Response
+     * @return mixed
      */
-    public function show($id)
+    public function show(Request $request)
     {
-        $id = $id;
+        $user = $request->user();
+
+        if (!$user) {
+            return response()->json(['success' => false], Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+
+        return response()->json(['data' => $user,'success' => true], Response::HTTP_OK);
```

Comment:
Server get user data from the request of client and response this?
`$user = $request->user();` and `return response()->json(['data' => $user,'success' => true], Response::HTTP_OK);`

Reply:
after the request through the Authentication passport with request access token, remove the access token from the request and replace with map users model, instead of use `Auth::guard('api')->user()`, i use `$request->user()`

Reply:
after request through the Oauth2 passport with access token. the $request->user() get the user map with the token